src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -256,7 +256,7 @@ public String produceDts(Depgraph depgraph) {
   private Set<String> getShadowedProvides(TreeSet<String> provides) {
     Set<String> shadowedProvides = new TreeSet<>();
     for (String provide : provides) {
-      if (!provides.subSet(provide + ".", provide + "\uFFFF").isEmpty()) {
+      if (!provides.subSet(provide + ".", provide + ".\uFFFF").isEmpty()) {
```

Comment:
Sometimes for this sort of thing I'll just sneak it in to an existing regression test.  You could write a unit test for this, but sometimes that sort of unit test will only check for exactly this bug and not whatever the next bug is.

I guess, use your judgement.  :)